src/workerd/api/trace.h

```diff
@@ -209,6 +211,25 @@ class TraceItem::QueueEventInfo final: public jsg::Object {
   const Trace::QueueEventInfo& eventInfo;
 };
 
+class TraceItem::EmailEventInfo final: public jsg::Object {
+public:
+  explicit EmailEventInfo(kj::Own<Trace> trace, const Trace::EmailEventInfo& eventInfo);
+
+  kj::StringPtr getMailFrom();
+  kj::StringPtr getRcptTo();
+  uint32_t getRawSize();
+
+  JSG_RESOURCE_TYPE(EmailEventInfo) {
+    JSG_READONLY_INSTANCE_PROPERTY(mailFrom, getMailFrom);
+    JSG_READONLY_INSTANCE_PROPERTY(rcptTo, getRcptTo);
+    JSG_READONLY_INSTANCE_PROPERTY(rawSize, getRawSize);
```

Comment:
Let's make these `JSG_READONLY_PROTOTYPE_PROPERTY`. Recent changes to garbage collection handling make instance properties significantly less desirable.

Reply:
Done.